### PR TITLE
merge datagov into master

### PR DESCRIPTION
**Pull request by rshewitt**
- related to [#4923](https://github.com/GSA/data.gov/issues/4923)

- datagov is adding the following features to master
  - dcatus writer
  - iso19115-3 reader
  - iso19115-2 reader

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,84 +1,137 @@
...
+    tzinfo (2.0.6)
+      concurrent-ruby (~> 1.0)
+    unicode-display_width (2.5.0)
     uuidtools (2.2.0)
 
 PLATFORMS
-  arm64-darwin-22
-  x86_64-linux
+  arm64-darwin-23
+  x86_64-darwin-23
```
Comment (btylerburton) [inline Gemfile.lock:126]: we might need to re-add the `x86_64-linux` platform back in to get it build in CI.

File: lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_format.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+                  # MD Format (required)
+                  xFormat = xDistFormat.xpath(@@formatXPath)[0]
+
+                  if xFormat.nil?
+                     msg = 'WARNING: ISO19115-2 reader: element \'gmd:MD_Format\' '\
+                        'is missing in \'gmd:distributionFormat\''
+                     hResponseObj[:readerExecutionMessages] << msg
+                     hResponseObj[:readerExecutionPass] = false
+                     return hFormat
```
Comment (btylerburton) [inline lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_format.rb:25]: I'd like some more clarity from Jonathan about when to return nil vs an empty metadata object.

**Review by btylerburton — APPROVED** [review]
I'm of the mind that we should be guided by our tests as this is just too massive to do anything but a cursory review of. 

Question though: would it make more sense to merge the DCAT-US base branch into master and then merge our stuff in on top of that, so we can easily see what we've changed outside the v2 / v3 readers.

**Comment by rshewitt** [comment]
> I'm of the mind that we should be guided by our tests as this is just too massive to do anything but a cursory review of.
> 
> Question though: would it make more sense to merge the DCAT-US base branch into master and then merge our stuff in on top of that, so we can easily see what we've changed outside the v2 / v3 readers.

i think this is a good idea. i'll do this.

File: lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_format.rb (hunk 2 above)
Comment (rshewitt) [reply lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_format.rb:25]: i agree. i have this as an improvement in the -2 reader [ticket](https://github.com/GSA/data.gov/issues/4862#:~:text=determine%20which%20modules%20must%20return%20hash%20vs%20nil%20comment1%2C%20comment2)

**Pull request closed without merging**